index.d.ts

```diff
@@ -0,0 +1,144 @@
+import EventEmitter, { EventMap } from 'bare-events'
+import { Duplex, DuplexEvents } from 'bare-stream'
+
+interface PipeEvents extends DuplexEvents {
+  connect: []
+}
+
+interface PipeOptions {
+  readBufferSize?: number
+  allowHalfOpen?: boolean
+}
+
+interface PipeConnectOptions {
+  path?: string
+}
+
+declare interface Pipe<M extends PipeEvents = PipeEvents> extends Duplex<M> {
+  readonly connecting: boolean
+  readonly pending: boolean
+  readonly readyState: 'open' | 'readOnly' | 'writeOnly'
+
+  open(fd: number, opts?: { fd?: number }, onconnect?: () => void): this
+
+  open(fd: number, onconnect: () => void): this
+
+  open(opts: { fd: number }, onconnect?: () => void): this
+
+  connect(path: string, opts?: PipeConnectOptions, onconnect?: () => void): this
+
+  connect(path: string, onconnect: () => void): this
+
+  connect(opts: PipeConnectOptions, onconnect?: () => void): this
+
+  ref(): void
+
+  unref(): void
+}
+
+declare class Pipe<M extends PipeEvents = PipeEvents> extends Duplex<M> {
+  constructor(path: string | number, opts?: PipeOptions)
+
+  constructor(opts?: { readBufferSize?: number; allowHalfOpen?: boolean })
+}
+
+interface ServerEvents extends EventMap {
+  close: []
+  connection: [pipe: Pipe]
+  err: [err: Error]
+  listening: []
+}
+
+interface ServerOptions {
+  readBufferSize?: number
+  allowHalfOpen?: boolean
+}
+
+declare interface Server<M extends ServerEvents = ServerEvents>
+  extends EventEmitter<M> {
+  readonly listening: boolean
+
+  address(): string | null
+
+  listen(
+    path: string,
+    backlog?: number,
+    opts?: { path?: string; backlog?: number },
```

Comment:
```suggestion
    opts?: { path?: string; backlog?: number },
```

Let's add a `ServerListenOptions` interface for this.